src/app/components/display-merged-run/display-merged-run.component.ts

```diff
@@ -60,4 +60,16 @@ export class DisplayMergedRunComponent implements OnDestroy {
             drawingArea.removeChild(drawingArea.lastChild as ChildNode);
         }
     }
+
+    public resetDrawing(): void {
```

Comment:
Das Zurücksetzen der Anzeige zum Sugiyama müssen wir denke ich anders umsetzen.
In der Funktion layout() sollten die manuell hinterlegten Koordinaten meiner Meinung nach nicht zurückgesetzt werden, ansonsten würden die Koordinaten immer verloren gehen, wenn sich das Input-Feld ändert.
Für das Zurücksetzen der manuell hinterlegen Koordinaten würde ich eine eigene Funktion (ggf. im Display-Service) anlegen. Können wir aber sonst auch gerne morgen drüber sprechen.

Reply:
vllt können wir auch in den elements zwei x und zwei y koordinaten speichern? eine für das freie hin und herschieben und die andere für die im sugiyama algorithmus berechnete position?

Reply:
Generell.
Wenn ich das richtig verstehe, soll über den Aufruf von this.layoutService.layout(run) hier das Layout komplett zurückgesetzt werden oder?
Ich denke, dass wir dafür eine separate Funktion brauchen, die manuelle Koordinaten löscht.

Reply:
aber wird nicht genau das  in der Funktion gemacht? die x und y attribute der elements gesetzt?

Reply:
Ja, aber sobald wir den Parser anpassen und die manuelle Positionierung speichern wollen, dürfen wir die diese X und Y Attribute in der layout-Funktion nicht mehr überschreiben. Sonst würden die jedes Mal verloren gehen, wenn das Textfeld sich ändert und das Layout neu berechnet wird. Das Layout sollt dann nur für die Transitions ohne manuelle Positionierung neu berechnet werden.

Die Reset-Funktion müsste denke ich das Textfeld aktualisieren und dort die manuell hinterlegten Koordinaten löschen, dann wird der Run neu geparst ohne Koordinaten-Informationen und alles wird wieder nach Sugiyama berechnet.

Reply:
hab einen neuen branch gemacht 39b und dort im Textfeld die Sachen gelöst, kannst du mal gucken ob das so theoretisch passen würde? Dann würde ich die Formatierung anpassen und einen PR machen.